src/vivarium/framework/engine.py

```diff
@@ -259,7 +260,7 @@ def step(self) -> None:
         self._clock.step_forward(self.get_population().index)
 
     def run(self) -> None:
-        while self._clock.time < self._clock.stop_time:
+        while not self.past_stop_time():
```

Comment:
I think it's better to put all of the backup logic here rather than in VCT. Then VCT can just call `sim.run(backup_path, backup_freq)`, where these values are None if we aren't backing up.

We can also get rid of `past_stop_time` if we do that.

Reply:
how's this?